
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Welcome to CUBAP’s documentation! &#8212; CUBAP  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="welcome-to-cubap-s-documentation">
<h1>Welcome to CUBAP’s documentation!<a class="headerlink" href="#welcome-to-cubap-s-documentation" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
</div>
<div class="section" id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</div>
<div class="section" id="codon-frequency">
<h1>Codon Frequency<a class="headerlink" href="#codon-frequency" title="Permalink to this headline">¶</a></h1>
<div class="section" id="graphs">
<h2>Graphs:<a class="headerlink" href="#graphs" title="Permalink to this headline">¶</a></h2>
<div class="section" id="average-codon-frequency">
<h3><em>Average Codon Frequency</em><a class="headerlink" href="#average-codon-frequency" title="Permalink to this headline">¶</a></h3>
<p>For the selected gene(s)/isoform(s), the average number of times each codon occurs across all samples.</p>
</div>
<div class="section" id="standard-deviation-of-frequency">
<h3><em>Standard Deviation of Frequency</em><a class="headerlink" href="#standard-deviation-of-frequency" title="Permalink to this headline">¶</a></h3>
<p>For the selected gene(s)/isoform(s), the standard deviation of the number of times each codon occurs across all samples.</p>
</div>
<div class="section" id="average-codon-frequency-by-superpopulation">
<h3><em>Average Codon Frequency by Superpopulation</em><a class="headerlink" href="#average-codon-frequency-by-superpopulation" title="Permalink to this headline">¶</a></h3>
<p>This violin plot shows the frequency of a certain codon, in the selected gene(s)/isoform(s), across all populations. Thicker sections represent a greater number of samples that contain that number of codon occurances.</p>
</div>
<div class="section" id="average-frequency-by-population">
<h3><em>Average Frequency by Population</em><a class="headerlink" href="#average-frequency-by-population" title="Permalink to this headline">¶</a></h3>
<p>This graph shows the same information as the violin plot except individual subpopulation averages are shown. Each point represents the average frequency of the selected codon for all samples in that subpopulation.</p>
</div>
</div>
<div class="section" id="options">
<h2>Options:<a class="headerlink" href="#options" title="Permalink to this headline">¶</a></h2>
<div class="section" id="select-gene">
<h3><em>Select Gene:</em><a class="headerlink" href="#select-gene" title="Permalink to this headline">¶</a></h3>
<p>Use the search and dropdown box features to find your gene(s) of interest. Click on the gene name to query it. You can query multiple genes. You must next select an isoform for every gene you’ve selected in order to view the data in the graphs.</p>
</div>
<div class="section" id="select-isoform">
<h3><em>Select Isoform</em><a class="headerlink" href="#select-isoform" title="Permalink to this headline">¶</a></h3>
<p>Click the dropdown arrow to view all the isoforms for the gene(s) you’ve already selected. To select multiple either use the ‘CTRL’ or ‘command’ button. You may also select all isoforms by clicking the ‘Select All’ button</p>
</div>
<div class="section" id="view-options">
<h3><em>View Options</em><a class="headerlink" href="#view-options" title="Permalink to this headline">¶</a></h3>
<p>If you have queried multiple genes or isoforms, you can view the data in two different ways. You can select the ‘Compare Genes’ button in order to view codon frequencies of different genes side by side. You can also select the “Average of Genes” button in order to view the average codon frequencies across all selected genes/isoforms. By default, the ‘Average of Genes’ view option is selected. These view options only affect the <em>Average Codon Frequency</em> and <em>Standard Deviation of Frequency</em> graphs.</p>
</div>
<div class="section" id="select-population">
<h3><em>Select Population</em><a class="headerlink" href="#select-population" title="Permalink to this headline">¶</a></h3>
<p>Choose a super= or subpopulation to view only the codon frequency data from those populations. This will filter every graph.</p>
</div>
<div class="section" id="select-codon">
<h3><em>Select Codon</em><a class="headerlink" href="#select-codon" title="Permalink to this headline">¶</a></h3>
<p>Choose a codon of interest to view how the frequency of this codon, in the selected gene(s)/isoform(s), across all populations. This only alters the <em>Average Codon Frequency by Superpopulation</em> and <em>Average Frequency by Population</em> graphs.</p>
</div>
</div>
</div>
<div class="section" id="identical-codon-pairing-and-co-trna-codon-pairing">
<h1>Identical Codon Pairing and Co=tRNA Codon Pairing<a class="headerlink" href="#identical-codon-pairing-and-co-trna-codon-pairing" title="Permalink to this headline">¶</a></h1>
<p>These visuals are set up identically to the <em>Codon Frequency</em> visual except that instead of showing the frequency of each codon, the frequency of each codon pair is shown. For co=tRNA codon pairing, synonymous (but not identical!) pairs are shown by their common amino acid.</p>
</div>
<div class="section" id="codon-aversion">
<h1>Codon Aversion<a class="headerlink" href="#codon-aversion" title="Permalink to this headline">¶</a></h1>
<div class="section" id="id1">
<h2>Graphs:<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<div class="section" id="total-codon-aversion-across-all-genes-by-superpopulation">
<h3><em>Total Codon Aversion Across all Genes by Superpopulation</em><a class="headerlink" href="#total-codon-aversion-across-all-genes-by-superpopulation" title="Permalink to this headline">¶</a></h3>
<p>This graph shows how often each codon is averted. Specifically, it is the total number of alleles, summed for each gene, in which the codon is not present. This total allele count is separated for each codon.</p>
</div>
<div class="section" id="total-number-of-alleles-per-superpopulation-with-codon">
<h3><em>Total Number of Alleles per Superpopulation with Codon</em><a class="headerlink" href="#total-number-of-alleles-per-superpopulation-with-codon" title="Permalink to this headline">¶</a></h3>
<p>Across the X=axis is the codon aversion motif (all the codons that are missing in the selected gene(s)/isform(s)) and the number of alleles, per superpopulation, that are missing that codon.</p>
</div>
</div>
<div class="section" id="id2">
<h2>Options:<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id3">
<h3><em>Select Gene:</em><a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p>Use the search and dropdown box features to find your gene(s) of interest. Click on the gene name to query it. Y
ou can query multiple genes. You must next select an isoform for every gene you’ve selected in order to view the data in the graphs.</p>
</div>
<div class="section" id="id4">
<h3><em>Select Isoform</em><a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<p>Click the dropdown arrow to view all the isoforms for the gene(s) you’ve already selected. To select multiple ei
ther use the ‘CTRL’ or ‘command’ button. You may also select all isoforms by clicking the ‘Select All’ button</p>
</div>
<div class="section" id="compare-subpopulations">
<h3><em>Compare Subpopulations</em><a class="headerlink" href="#compare-subpopulations" title="Permalink to this headline">¶</a></h3>
<p>Click this button to view the number of alleles that are missing codons by subpopulation instead of superpopulation. This only affects the <em>Total Number of Alleles per Superpopulation with Codon</em> graph. Click the <em>Reset</em> button to view superpopulation data again.</p>
</div>
<div class="section" id="select-subpopulations">
<h3><em>Select Subpopulations</em><a class="headerlink" href="#select-subpopulations" title="Permalink to this headline">¶</a></h3>
<p>This filter only applies to the <em>Total Number of Alleles per Subpopulation with Codon</em> graph. It allows you to ivew the number of alleles of only certain subpopulations.</p>
</div>
</div>
</div>
<div class="section" id="ramp-sequences">
<h1>Ramp Sequences<a class="headerlink" href="#ramp-sequences" title="Permalink to this headline">¶</a></h1>
<div class="section" id="id5">
<h2>Graphs:<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h2>
<div class="section" id="ramp-harmonic-mean-rscu-by-subpopulation">
<h3><em>Ramp Harmonic Mean RSCU by Subpopulation</em><a class="headerlink" href="#ramp-harmonic-mean-rscu-by-subpopulation" title="Permalink to this headline">¶</a></h3>
<p>For all selected gene(s)/isoform(s), the harmonic mean of all RSCU values for each codon in the ramp sequence. This is plotted in a box and whiskers plot by subpopulation.</p>
</div>
<div class="section" id="gene-harmonic-mean-rscu-bu-subpopulation">
<h3><em>Gene Harmonic Mean RSCU bu Subpopulation</em><a class="headerlink" href="#gene-harmonic-mean-rscu-bu-subpopulation" title="Permalink to this headline">¶</a></h3>
<p>For all selected gene(s)/isoform(s), the harmonic mean of all RSCU values for each codon in the entire gene sequence. This is plotted in a box and whiskers plot by subpopulation.</p>
</div>
</div>
<div class="section" id="averages">
<h2>Averages:<a class="headerlink" href="#averages" title="Permalink to this headline">¶</a></h2>
<p>Ramp Sequence
= RSCU: the average RSCU value for all codons in the ramp sequence, from all populations
= Length: the average length of the ramp sequence, in number of codons, from all populations
Entire Gene
= RSCU: the average RSCU value for all codons in the entire gene sequence, from all populations
= Length: the average length of the entire gene sequence, in number of codons, from all populations</p>
<div class="section" id="if-multiple-genes-isoforms-are-selected-these-values-will-also-be-the-averages-of-all-those">
<h3>If multiple genes/isoforms are selected, these values will also be the averages of all those.<a class="headerlink" href="#if-multiple-genes-isoforms-are-selected-these-values-will-also-be-the-averages-of-all-those" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="id6">
<h2>Options:<a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id7">
<h3><em>Select Gene</em><a class="headerlink" href="#id7" title="Permalink to this headline">¶</a></h3>
<p>Use the search and dropdown box features to find your gene(s) of interest. Click on the gene name to query it. Y
ou can query multiple genes. You must next select an isoform for every gene you’ve selected in order to view the data in the graphs.</p>
</div>
<div class="section" id="id8">
<h3><em>Select Isoform</em><a class="headerlink" href="#id8" title="Permalink to this headline">¶</a></h3>
<p>Click the dropdown arrow to view all the isoforms for the gene(s) you’ve already selected. To select multiple ei
ther use the ‘CTRL’ or ‘command’ button. You may also select all isoforms by clicking the ‘Select All’ button</p>
</div>
</div>
</div>
<div class="section" id="nucleotide-composition">
<h1>Nucleotide Composition<a class="headerlink" href="#nucleotide-composition" title="Permalink to this headline">¶</a></h1>
<div class="section" id="id9">
<h2>Graphs:<a class="headerlink" href="#id9" title="Permalink to this headline">¶</a></h2>
<div class="section" id="average-nulceptide-frequency">
<h3><em>Average Nulceptide Frequency</em><a class="headerlink" href="#average-nulceptide-frequency" title="Permalink to this headline">¶</a></h3>
<p>The average number of occurances of each nucleotide in the selected gene(s)/isoform(s).</p>
</div>
<div class="section" id="standard-deviation-of-nucleotide-frequency">
<h3><em>Standard Deviation of Nucleotide Frequency</em><a class="headerlink" href="#standard-deviation-of-nucleotide-frequency" title="Permalink to this headline">¶</a></h3>
<p>The standard deviation of the number of occurances of each nucleotide in the selected gene(s)/isoform(s).</p>
</div>
<div class="section" id="gc-content-across-populations">
<h3><em>GC Content Across Populations</em><a class="headerlink" href="#gc-content-across-populations" title="Permalink to this headline">¶</a></h3>
<p>A violin plot of GC content compared across Superpopulations.</p>
</div>
<div class="section" id="nucleotide-frequency-by-superpopulation">
<h3><em>Nucleotide Frequency by Superpopulation</em><a class="headerlink" href="#nucleotide-frequency-by-superpopulation" title="Permalink to this headline">¶</a></h3>
<p>The frequency of the selected nucleotide for each superpopulation.</p>
</div>
<div class="section" id="nucleotide-frequency-by-superpopulation-and-subpopulation">
<h3><em>Nucleotide Frequency by Superpopulation and Subpopulation</em><a class="headerlink" href="#nucleotide-frequency-by-superpopulation-and-subpopulation" title="Permalink to this headline">¶</a></h3>
<p>The frequency of the selected nulceotide for each subpopulation, ordered by superpopulation.
The <em>Average GC Content %</em> is also shown. This is computed from all selected genes/isoforms and from all populations.</p>
</div>
</div>
<div class="section" id="id10">
<h2>Options:<a class="headerlink" href="#id10" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id11">
<h3><em>Select Gene</em><a class="headerlink" href="#id11" title="Permalink to this headline">¶</a></h3>
<p>Use the search and dropdown box features to find your gene(s) of interest. Click on the gene name to query it. Y
ou can query multiple genes. You must next select an isoform for every gene you’ve selected in order to view the data in the graphs.</p>
</div>
<div class="section" id="id12">
<h3><em>Select Isoform</em><a class="headerlink" href="#id12" title="Permalink to this headline">¶</a></h3>
<p>Click the dropdown arrow to view all the isoforms for the gene(s) you’ve already selected. To select multiple ei
ther use the ‘CTRL’ or ‘command’ button. You may also select all isoforms by clicking the ‘Select All’ button</p>
</div>
<div class="section" id="gc-content">
<h3><em>GC Content</em><a class="headerlink" href="#gc-content" title="Permalink to this headline">¶</a></h3>
<p>This is selected by default. It shows the <em>GC Content Across Populations</em> graph.</p>
</div>
<div class="section" id="nucleotide-frequenc">
<h3><em>Nucleotide Frequenc</em><a class="headerlink" href="#nucleotide-frequenc" title="Permalink to this headline">¶</a></h3>
<p>Click this button to view the <em>Nucleotide Frequency by Superpopulation</em> and <em>Nucleotide Frequency by Superpopulation and Subpopulation</em> graphs.</p>
</div>
</div>
</div>
<div class="section" id="frequently-asked-questions">
<h1>Frequently Asked Questions<a class="headerlink" href="#frequently-asked-questions" title="Permalink to this headline">¶</a></h1>
<p><em>Why are some of the graphs blank?</em>
Because these graphs compare population differences, but the selected gene(s)/isoform(s) only have one allele in the entire 1000 Genome Sample and thus do not differ at all between populations.</p>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="#">CUBAP</a></h1>








<h3>Navigation</h3>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2020, Matthew Hodgman, Justin Miller, Taylor Meurs, John Kauwe.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 2.2.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>